en/flight_stack/controller_diagrams.md

```diff
@@ -16,6 +16,62 @@ Request access from dev team. -->
 * Depending on the mode, the outer (position) loop is bypassed (shown as a multiplexer after the outer loop). The position loop is only used when holding position or when the requested velocity in an axis is null.
 * The integrator in the inner loop (velocity) controller includes an anti-reset windup (ARW) using a clamping method.
 
+## Fixed-Wing Position Controller
+
+### Total Energy Control System (TECS)
+The PX4 implementation of the Total Energy Control System (TECS) enables simultaneous control of true airspeed and altitude of a fixed wing aircraft.
+The code is implemented as a library which is used in the fixed wing position control module (make a reference here). As seen in the diagram below, TECS receives as inputs airspeed and altitude setpoints and outputs a throttle and pitch angle setpoint. These two outputs are sent to the fixed wing attitude controller which implements the attitude control solution. It's therfore important to understand that the performance of TECS is directly affected by the performance of the pitch control loop. A poor tracking of airspeed and altitude is often caused by a poor tracking of the aircraft pitch angle.
+
+> **Note** Make sure to tune the attitude controller before attempting to tune TECS.
```

Comment:
Good comment. Is there a link to tuning the attitude controller? - or is this the doc you still plan to write?

Reply:
@hamishwillee Not sure if there is. If yes, I will add it. Otherwise it's something for the future.

Reply:
OK then. I'm going to merge this now!